### PR TITLE
Use different Bugzilla comment when translation is missing

### PR DESCRIPTION
**Pull request by enderandpeter**
#827 

# The Problem

After clicking the "report a bug" link for a missing target translation, the Bugzilla comment contains the **@@missing@@** tag used internally by this application.

# The Solution

When the Bugzilla comment is created for the "report a bug" link, and the target translation is missing, use a default comment that pertains to this situation, for example:

```
Source string:
Content crashed

This string has not been translated yet. Proposed translation:
```

# Notes

I have a simple solution in place that will use the above template if the `$target_string` in `Bugzilla::reportErrorLink` is `@@missing@@`. `tests/units/Transvision/Bugzilla.php` has been updated to test for this case. See if you can think of other cases we may want to test.

This is my first contribution to this project, so I apologize if I missed something. I did run `php-cs-fixer` on the changes files. I hope the other tests still pass, but I will do what is necessary to fix any problems. Let me know what you think. Thank you!

### CODE REVIEW

File: app/classes/Transvision/Bugzilla.php [hunk 1]
```diff
@@ -55,14 +55,26 @@ public static function reportErrorLink($locale, $entity, $source_string, $target
...
+
+        $default_message = "The string:\n{$source_string}\n\n"
+            . "Is translated as:\n{$target_string}\n\n"
+            . "And should be:\n\n\n\n"
+            . "Feedback via Transvision:\n"
+            . $transvision_url;
+
+        $message = $default_message;
+
+        // If the target string has the @@missing@@ tag, use an alternate comment
```
Comment (flodolo) [inline app/classes/Transvision/Bugzilla.php:67]:
```suggestion
        // If the target string has the @@missing@@ tag, use an alternative comment
```

**Review by flodolo — CHANGES_REQUESTED** [review]
This looks great and works, but it also made me realize there is more space for improvement. In the almost 10 years since this code was originally written, Bugzilla has introduced support for Markdown, which can help us a lot here.

We can use code blocks for source/translation, and an actual link instead of the huge Transvision URL. Also, to simplify code a bit, we can use the same preamble (`Source string:`) for both scenarios.

Example:

```php
$message = "Source string:\n\n```\n{$source_string}\n```\n\n";
if($target_string !== '@@missing@@') {
    $message .= "Is translated as:\n\n```\n{$target_string}\n```\n\n"
        . "And should be:\n\n```\n(add your translation here)\n```\n\n";
} else {
    // If the target string has the @@missing@@ tag, use an alternative comment
    $message .= "This string has not been translated yet. Proposed translation:\n\n"
        . "```\n(add your translation here)\n```\n\n";
}
// Add link to the string in Transvision
$message .= "Feedback via [Transvision]({$transvision_url}).";
$bug_message = rawurlencode(
    html_entity_decode($message)
);
```

Feel free to change/improve it, but that should be the general idea (only tested with a couple of strings).

**Comment by enderandpeter** [comment]
I was able to incorporate the changes you suggested. Things appear to be working as expected. Here are some previews of the Bugzilla comment:

## New comment, string missing

New Bugzilla Comment_String Missing

## String Not Missing

New Bugzilla Comment_String Not Missing

## Complex String Missing

New Bugzilla Comment_Complex String Missing

## Complex String Not Missing

New Bugzilla Comment_Complex String Not Missing

I'm wondering if the newlines insides the preformatted blocks should be removed? It appears they are creating several spaces instead of an actual newline.

**Comment by flodolo** [comment]
> Actually, I do see the newlines now. I think the indentation threw me off. Not sure where that is from.

Not sure if [this](https://transvision.flod.org/?sourcelocale=en-US&locale=af&repo=all_projects&search_type=entities&recherche=android_l10n/mozilla-mobile/android-components/components/browser/errorpages/src/main/res/values/strings.xml:mozac_browser_errorpages_safe_browsing_malware_uri_message&entire_string=entire_string) is the string in your second example, but there are spaces in the source

```
    <string name="mozac_browser_errorpages_safe_browsing_malware_uri_message"><![CDATA[
      <p>The site at %1$s has been reported as an attack site and has been blocked based on your security preferences.</p>
    ]]></string>
```

If you inspect the results table, and set `white-space: pre`, you'll see them.

Screenshot 2023-04-10 alle 07 35 14

**Review by flodolo — APPROVED** [review]
Congrats for your first contribution.

This will be available in a few minutes in my own server (running beta), and deployed in production in the coming days (hopefully we can figure out the issue with All Projects and deploy both).

**Comment by enderandpeter** [comment]
Heck yeah. Thank you so much!

**Pull request merged**